portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -1074,6 +1077,98 @@ public DLFileEntry fetchFileEntryByName(
 		return dlFileEntryPersistence.fetchByG_F_N(groupId, folderId, name);
 	}
 
+	@Override
+	public void forEachFileEntry(
+			long companyId, Consumer<DLFileEntry> consumer, long maximumSize,
+			String[] mimeTypes)
+		throws PortalException {
+
+		forEachFileEntry(companyId, 0, consumer, maximumSize, mimeTypes);
+	}
+
+	@Override
+	public void forEachFileEntry(
+			long companyId, long classNameId, Consumer<DLFileEntry> consumer,
+			long maximumSize, String[] mimeTypes)
+		throws PortalException {
+
+		_performDynamicQueryActions(
+			consumer,
+			dynamicQuery -> {
+				Property companyIdProperty = PropertyFactoryUtil.forName(
+					"companyId");
+				Property mimeTypeProperty = PropertyFactoryUtil.forName(
+					"mimeType");
+
+				dynamicQuery.add(companyIdProperty.eq(companyId));
+				dynamicQuery.add(mimeTypeProperty.in(mimeTypes));
+
+				if (classNameId > 0) {
+					Property classNameIdProperty = PropertyFactoryUtil.forName(
+						"classNameId");
+
+					dynamicQuery.add(classNameIdProperty.eq(classNameId));
+				}
+
+				if (maximumSize >= 0) {
+					Property sizeProperty = PropertyFactoryUtil.forName("size");
+
+					dynamicQuery.add(sizeProperty.le(maximumSize));
+				}
+			});
+	}
+
+	@Override
+	public void forEachFileEntryNotInTrash(
```

Comment:
Instead of creating a method named like this, there are two customary ways to exclude entries in trash:
1. Add a boolean `includeTrashedEntries` parameter.
2. Use `QueryDefinition` (doesn't make much sense here, as this is not returning results).

There are some examples of this kind of usage in this class itself.